model_zoo/census_model_sqlflow/dnn/census_dnn.sql

```diff
@@ -0,0 +1,19 @@
+SELECT *
+FROM census_income
+TO TRAIN DNNClassifier
+WITH model.hidden_units = [10, 20]
+COLUMN (
+    age, 
+    capital_gain, 
+    capital_loss, 
+    hours_per_week, 
+    EMBEDDING(HASH(workclass, 64), 16),
```

Comment:
I think hash is part of the embedding operation. How about we define only one data transform function `embedding`:

```sql
COLUMN
    embedding(workclass, 64, 16),
...
```

Reply:
Sometimes user will call `lookup vocabulary` to convert workclass from string to id, and then look up the embedding.
Maybe we need add a flag to differentiate hash or look_up_vocabulary for the id conversion.

Reply:
We actually have https://www.tensorflow.org/api_docs/python/tf/feature_column/categorical_column_with_hash_bucket as a feature column. The problem is how to calculate the bucket size (64 here) using some SQL statement and pass it here.

Reply:
We can use the following SQL statement to calculate the hash_bucket_size for `workclass`. `3` is  a configurable factor to decrease the conflict of the hash function.
SELECT 
    (COUNT(DISTINCT(workclass)) * 3) AS _workclass_hash_bucket_num_
FROM census_income;

For more details for analysis SQL generation, we can see #1667

Reply:
> @brightcoder01 Let us use `embedding(workclass, 64, 16)` instead of `embedding(hash(workclass, 64), 16)`. If `embedding` creates a `map[int]Vector`, it can call `hash` to convert `workclass` into the integer ID.

Good idea. The SQLFlow expression describes users' intention and don't need expose the details about how to convert string to id. We will use embedding(workclass, dimension=16) directly.